src/main_benchmark.cpp

```diff
@@ -443,7 +443,7 @@ int main(int argc, char** argv) {
     size_t nireq = cliparser.result->operator[]("nireq").as<uint32_t>();
     size_t niter = cliparser.result->operator[]("niter").as<uint32_t>();
     size_t threadsPerIreq = cliparser.result->operator[]("threads_per_ireq").as<uint32_t>();
-    size_t threadCount = nireq * threadsPerIreq;
+    size_t threadCount = std::min(nireq * threadsPerIreq, niter);
```

Comment:
Write in description what was the issue.

Why previous logic didn't work? We may want add assertions in corner cases if this doesn't work. Otherwise setting nireq & threadsPerIreq by user does not work.